src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -460,18 +461,28 @@ public void setupIoTRoleForTes(String roleName, String roleAliasName, String cer
             return Optional.of(iamClient.getPolicy(software.amazon.awssdk.services.iam.model.GetPolicyRequest.builder()
                     .policyArn(String.format(MANAGED_IAM_POLICY_ARN_FORMAT, partition, policyName)).build()).policy()
                     .arn());
-        } catch (ResourceNotFoundException | NoSuchEntityException mnf) {
-            // Check if a customer policy exists with the name
-            try {
-                return Optional.of(iamClient.getPolicy(
-                        software.amazon.awssdk.services.iam.model.GetPolicyRequest.builder()
-                                .policyArn(String.format(IAM_POLICY_ARN_FORMAT, partition, getAccountId(), policyName))
-                                .build())
-                        .policy().arn());
-            } catch (ResourceNotFoundException | NoSuchEntityException cnf) {
-                return Optional.empty();
+        } catch (NoSuchEntityException mnf) {
+            outStream.println("No managed IAM policy found, looking for user defined policy...");
+        } catch (IamException e) {
+            if (e.getMessage().contains("not authorized to perform")) {
+                outStream.println("No permissions to lookup managed policy, looking for user defined policy...");
+            }
```

Comment:
Do we need to log other IamExceptions? same below

Reply:
I can do that, but since we can't do anything about those exceptions and script will fail anyway so the logging might be redundant